pkg/build/admission/defaults/admission_test.go

```diff
@@ -130,3 +130,143 @@ func TestIncrementalDefaults(t *testing.T) {
 	}
 
 }
+func TestResourceDefaults(t *testing.T) {
+	tests := map[string]struct {
+		DefaultResource  kapi.ResourceRequirements
+		BuildResource    kapi.ResourceRequirements
+		ExpectedResource kapi.ResourceRequirements
+	}{
+		"BuildDefaults plugin and BUILD object both defined resource limits and requests": {
+			DefaultResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
```

Comment:
for this and all the other tests, use different values for "requests" and "limits" to ensure the values aren't getting mixed up.  As it is if you were assigning the limit value to the resource field, the test would still pass.